nautobot/core/utils/git.py

```diff
@@ -73,7 +73,10 @@ def __init__(self, path, url, clone_initially=True):
         elif clone_initially:
             # Don't log `url` as it may include authentication details.
             logger.debug("Cloning git repository to %s...", path)
-            self.repo = Repo.clone_from(url, to_path=path, env=GIT_ENVIRONMENT)
+            if not shallow:
+                self.repo = Repo.clone_from(url, to_path=path, env=GIT_ENVIRONMENT)
+            else:
+                self.repo = Repo.clone_from(url, to_path=path, env=GIT_ENVIRONMENT, depth=1)
```

Comment:
Wonder if it'd be any cleaner from an API perspective to implement a `depth` parameter instead of a `shallow` parameter? Alternatively:

```suggestion
            kwargs = {}
            if shallow:
                kwargs["depth"] = 1
            self.repo = Repo.clone_from(url, to_path=path, env=GIT_ENVIRONMENT, **kwargs)
```

Reply:
Implementing a `depth` parameter instead of `shallow` makes sense